lib/closure_tree/hierarchy_maintenance.rb

```diff
@@ -35,10 +35,13 @@ def _ct_before_save
     end
 
     def _ct_after_save
-      if changes[_ct.parent_column_name] || @was_new_record
+      as_5_1 = ActiveSupport.version >= Gem::Version.new('5.1.0')
+      changes_method = as_5_1 ? :saved_changes : :changes
```

Comment:
Yeah, maybe? I guess I'm just frustrated by the rails team for making changes like this. Over and over again. If you think the other way is better, I'll merge your PR.